protos/joystick/joystick.proto

```diff
@@ -0,0 +1,138 @@
+syntax = "proto3";
+
+package mavsdk.rpc.joystick;
+
+option java_package = "io.mavsdk.joystick";
+option java_outer_classname = "JoystickProto";
+
+service JoystickService
+{
+
+    rpc ConnectDrone (ConnectRequest)  returns (ConnectResponse) {}
+    /*
+     * Send command to enable Joystick mode 
+     *
+     * Enabling Joystic mode causes to change into Position Control mode
+     * drone still needs to be armed
+     * Needs to check if the drone is not currently flying or in mission if so drone needs to be paused first 
+     */
+    
+    rpc EnableJoystickMode (EnableRequest)  returns(EnableResponse) {}
+
+
+
+    rpc ArmDrone (ArmRequest) returns (ArmResponse) {}
+
+
+    rpc DisarmDrone (DisarmRequest) returns (DisarmResponse) {}
+
+      /*
+     * Send command to command to the drone to move forwards or backwards 
+     */
+    rpc Xmotion (XmotionRequest) returns (XmotionResponse) {}
+
+     /*
+     * Send command to command to the drone to move left or right 
+     */
+    rpc Ymotion (YmotionRequest) returns (YmotionResponse) {}
+
+     /*
+     * Send command to command to the drone to move up  or down 
+     */
+    rpc Zmotion (ZmotionRequest) returns (ZmotionResponse) {}
+
+     /*
+     * Send command to command to rotate clockwise or counterclockwise 
+     */
+    rpc Rmotion (RmotionRequest) returns (RmotionResponse) {}
```

Comment:
Do you really need specific requests for each axis or could it be one request containing all 4 axes at once?

Reply:
Assuming 2 joystick sticks, each stick will control 2 axis. 
I think it can be done all at once,  I just wanted a way to assign the axis to each stick, but I can see that it is better to put all of then at once.

Reply:
I am not super experienced with the [`MANUAL_CONTROL`](https://mavlink.io/en/messages/common.html#MANUAL_CONTROL) message, but when I used it for tests, I was sending one message with all the values, at 20Hz. At the MAVLink implementation level, I think we will need to aggregate those values into one message anyway, right? Otherwise the values that are not set will be interpreted at "0", won't they?

One question, though, is whether it should work like the offboard plugin (i.e. you set only the change of position of the joystick, and MAVSDK keeps the stick in that position until you actively change it), or if we should delegate the sending of those messages to the user (i.e. the MAVSDK user must send the message at some rate, e.g. 10-20Hz). Opinions there?

Reply:
I have push the new changes, also changing the name to manual control so it makes more sense

Reply:
Thinking about my point above, I think that the user will anyway have to repeatedly send the value when it changes, so we should delegate that (unlike the offboard plugin).

Reply:
I think it should a neutral position(hold position) in this case that is both sticks on the centre, and the message should be send as long it is move from the initial position, so when the initial position still there should not be a message sent, otherwise  move the sticks and hold them at position to obtain motion.  to stop motion sticks should be released. 

But you are right messages should be send when the stick changes position however, there should be a threshold to use +-10 depending on the input I do not know about the sensibility of the joystick and how those values are read.  there is not really a point on sending the same message 10 to 20 times in a second if they are all the same. 

should it be a state that detects that a joystick is active and if it is inactive it triggers a safemode ?

Reply:
> there is not really a point on sending the same message 10 to 20 times in a second if they are all the same

Are you sure about that? Do you already use `MANUAL_CONTROL` in your code? In my experience, the message has to be sent repeatedly no matter what, otherwise PX4 reports "data link lost" or something like that.

Reply:
I  havent actually implemented  it. thanking for letting me know